### ISSUE TITLE
no-unused-vars only detecting last parameter

### ISSUE THREAD

**Issue by BlaMaeda** [eslint/eslint#4292]
I have the following file:

``` javascript
'use strict';

/* eslint no-unused-vars:2 */

(function() {
angular
    .module('app')
    .controller('ctrl', ctrl);

function ctrl($scope, $q, $log, $mdDialog) {
}
})();
```

and eslint is only recognizing `$mdDialog` as unused. If I add a reference to it in the body of `ctrl`, that only error disappears, but the first three parameters (still unused) don't show up.

I'm using eslint v1.7.3.

Thanks.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
That's an expected behavior. If you are using 4th parameter, you need previous 3 parameters to get the right value. Even if they are unused, they serve a purpose.

**Comment by BlaMaeda** [comment]
Ohh, I see! Sorry, should have figured it on my own. Thanks!
